met-api/src/met_api/models/base_model.py

```diff
@@ -67,16 +67,10 @@ def flush(self):
         db.session.flush()
         return self
 
-    def add_to_session(self):
-        """Save and flush."""
-        return self.flush()
-
     def save(self):
         """Save and commit."""
-        self._set_tenant_id()
-        db.session.add(self)
-        db.session.flush()
-        db.session.commit()
+        self.flush()
```

Comment:
Does `self` access `db.session` in this instance?

Reply:
The first argument to a method of a class (called [`self`](https://www.w3schools.com/python/gloss_python_self.asp) by convention) always receives a reference to the class itself (in this case, `BaseModel`). So `self.flush()` is the same as [`BaseModel.flush()`](https://github.com/bcgov/met-public/blob/418208247e4ace26a8006cb3175db59ef048abc9/met-api/src/met_api/models/base_model.py#L63), which behind the scenes gets called as something like BaseModel.flush(self=BaseModel)